modules/core/src/lib/deck.js

```diff
@@ -717,7 +717,10 @@ export default class Deck {
     // If initialViewState was set on creation, auto track position
     if (this.viewState) {
       this.viewState[params.viewId] = viewState;
-      this.viewManager.setProps({viewState});
```

Comment:
Just some general thoughts (for future reference):
- This embedded view state business is not exactly easy to reason about. 
- I can't be sure on reviewing this code that it does what it should, because I am not 100% sure what cases it needs to handle.
- E.g. is this function called multiple times, once for each "embedded" view state? or is it sometimes called on top-level view state, and sometimes embedded view states depending on what is supplied?

What could be done to simplify this for future maintainers (and API users)?
- Call the field `this.viewStates` to make it more clear that internally we always store a map of view states
- Normalize a non-embedded view state to a `this.viewStates[`default`]`? (maybe we already do this.
- Potentially expose `props.viewStates` to remove the overload of `props.viewState` (view state vs map of view states)
- ...

Reply:
So let's define `<ViewState>` as a flat view props object, and `<ViewStateMap>` as an object that maps view ids to `<ViewState>` objects.

- `props.viewState` accepts either `<ViewState>` or `<ViewStateMap>`
- `this.viewState` is a `<ViewStateMap>`.
- The bug was that we used to update the `ViewManager` with a `<ViewState>` of whichever view that recently got updated. This PR changes to always update the `ViewManager` with a `<ViewStateMap>`.

I am reluctant to rename the prop because it's not clear how much we'd gain from the breaking change:

- Majority of deck.gl's use cases are still single view
- The name `viewStates` does not sufficiently communicate that it expects a map (plural prop names usually refer to array types).

Reply:
> The name viewStates does not sufficiently communicate that it expects a map (plural prop names usually refer to array types).

Yes, I believe this goes to the heart of the matter. I agree that it is hard to feel really comfortable with the map-based API. 

Idea: Given that view states can have `id`, we don't really need a map keyed on `id`s. Would it be possible to support an array of view states instead, and deprecate the map? 

As you say, this would make view states more orthogonal with the rest of the API. I suppose the downside would be that the user would need to do more work to manage the updates of such an array (i.e. the user might need to maintain a map separately for the updates)?

And initialViewState and onViewStateChange handling might need to be overhauled to support the new case.

Worth thinking through? Whether we would do this in 8.0 or later, it is always nice to have a sense of where we want each part of the API to go.